### PR TITLE
Clean up IOCell types and parameterization

### PR DESCRIPTION
**Pull request by jerryz123**

### CODE REVIEW

**Review by colinschmidt — APPROVED** [review]
LGTM

File: iocell/src/main/scala/chisel/IOCell.scala [hunk 1]
```diff
@@ -63,76 +63,68 @@ class DigitalInIOCellBundle extends Bundle {
   val ie = Input(Bool())
 }
 
-abstract class IOCell extends BlackBox
+trait IOCell extends BaseModule
 
-abstract class AnalogIOCell extends IOCell {
-  val io: AnalogIOCellBundle
```
Comment (jwright6323) [inline iocell/src/main/scala/chisel/IOCell.scala:69]: I don't think you should have changed this; it was done this way intentionally to allow the user to extend the IOCellBundle with foundry IO cell pins. Can you have multiple `IO`s in `BlackBox`es now?

File: iocell/src/main/scala/chisel/IOCell.scala [hunk 2]
```diff
@@ -63,76 +63,68 @@ class DigitalInIOCellBundle extends Bundle {
   val ie = Input(Bool())
 }
 
-abstract class IOCell extends BlackBox
+trait IOCell extends BaseModule
```
Comment (jwright6323) [inline iocell/src/main/scala/chisel/IOCell.scala:66]: Why does this need to be a `BaseModule`? The only way to drive `Analog` pins is a `BlackBox`.

**Review by jwright6323 — COMMENTED** [review]
A little late since this was merged already, but I think this actually regresses on some features and should be partially reverted.

File: iocell/src/main/scala/chisel/IOCell.scala (hunk 2 above)
Comment (jerryz123) [reply iocell/src/main/scala/chisel/IOCell.scala:66]: FireSim's IOCell models cannot be blackboxes

File: iocell/src/main/scala/chisel/IOCell.scala (hunk 1 above)
Comment (jerryz123) [reply iocell/src/main/scala/chisel/IOCell.scala:69]: I can add this feature back in

Comment (jwright6323) [reply iocell/src/main/scala/chisel/IOCell.scala:69]: Cool, thanks. I think everything else looks good, for the record.

Comment (jerryz123) [reply iocell/src/main/scala/chisel/IOCell.scala:69]: Actually, it seems like you can do this in the current system: 
```
class FancyDigitalGPIOCellBundle extends DigitalGPIOCellBundle {
  val extra = Output(Bool())
}

class FancyDigitalGPIOCell extends GenericIOCell with DigitalGPIOCell {
  override val io = IO(new FancyDigitalGPIOCellBundle)
}
```
so no changes are necessary?

Comment (jwright6323) [reply iocell/src/main/scala/chisel/IOCell.scala:69]: I don't think `override` works- you'll call `IO` twice and end up with phantom `IO`s that are disconnected (also I don't think you can call `IO` twice in `BlackBox` anyway)

Comment (jerryz123) [reply iocell/src/main/scala/chisel/IOCell.scala:69]: NVM the above doesn't work.

**Pull request merged**